datadog-iam-user-handler/src/datadog_iam_user/handlers.py

```diff
@@ -0,0 +1,183 @@
+import logging
+from typing import Any, MutableMapping, Optional
+
+from cloudformation_cli_python_lib import (
+    Action,
+    HandlerErrorCode,
+    OperationStatus,
+    ProgressEvent,
+    Resource,
+    SessionProxy,
+)
+from datadog_api_client.v1 import ApiException
+from datadog_api_client.v1.api import users_api
+from datadog_api_client.v1.model.access_role import AccessRole
+from datadog_api_client.v1.model.user import User
+from datadog_cloudformation_common.api_clients import v1_client
+
+from .models import ResourceHandlerRequest, ResourceModel
+from .version import __version__
+
+# Use this logger to forward log messages to CloudWatch Logs.
+LOG = logging.getLogger(__name__)
+TYPE_NAME = "Datadog::IAM::User"
+
+resource = Resource(TYPE_NAME, ResourceModel)
+test_entrypoint = resource.test_entrypoint
+
+
+@resource.handler(Action.CREATE)
+def create_handler(
+    session: Optional[SessionProxy],
+    request: ResourceHandlerRequest,
+    callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+    progress: ProgressEvent = ProgressEvent(
+        status=OperationStatus.IN_PROGRESS,
+        resourceModel=model,
+    )
+
+    LOG.info("Starting the User Resource Create Handler")
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL or "https://api.datadoghq.com",
+            TYPE_NAME,
```

Comment:
My mistake, let's not use `TYPE_NAME` directly here, and instead create a new constant with value `iam-user` to keep the same value as the java version https://github.com/DataDog/datadog-cloudformation-resources/blob/06c00b86d77c864001e36540844ed93519c800a6/datadog-iam-user-handler/src/main/java/com/datadog/iam/user/ClientFactory.java#L25